trio/_util.py

```diff
@@ -1,4 +1,5 @@
 # Little utilities we use internally
+from __future__ import annotations
```

Comment:
Should we preemptively stick this at the top of all files so that these kinds of change is already done?

Reply:
Actually IDK how I feel about using the cooler syntax early: we could use `typing.List` etc. and already-mentioned `pyupgrade` will drop that for newer syntax the moment we drop 3.8.

On the other hand it's more readable and more understandable. I guess `List[int]` vs `list[int]` (using `from typing import ...`) isn't too big a difference.

Reply:
The biggest downside to using `typing.List` is the additional overhead in import management - which is fully manual given our current lack of `isort`. This can be quite frustrating IME.

idc much whether to preemptively adding it

Reply:
I've liked using https://pypi.org/project/flake8-type-checking/ (or https://pypi.org/project/ruff/ ) for managing this.

Reply:
there's never a reason to not use ``from __future__ import annotations`` so you might as well use 3.11+ syntax wherever possible.

Reply:
fucking github. anyway there's never a reason not to use ``from __future__ import annotations`` so you might as well just use 3.11+ typing syntax always.

Reply:
https://github.com/Zac-HD/flake8-trio/pull/78#pullrequestreview-1208679518
https://github.com/Instagram/LibCST/issues/870
There's some scenarios where it matters, but I think we're fine.